src/Symfony/Component/HttpFoundation/File/MimeType/FileBinaryMimeTypeGuesser.php

```diff
@@ -43,7 +43,7 @@ public function __construct($cmd = 'file -b --mime %s 2>/dev/null')
      */
     public static function isSupported()
     {
-        return '\\' !== DIRECTORY_SEPARATOR && function_exists('passthru') && function_exists('escapeshellarg');
+        return '\\' !== DIRECTORY_SEPARATOR && function_exists('passthru') && function_exists('escapeshellarg') && null !== shell_exec('command -v file');
```

Comment:
since the performance penalty for doing this call is high, the result should be cached

Reply:
Done.